ejb/WFLY-10057-thread-pool.adoc

```diff
@@ -0,0 +1,223 @@
+= EJB subsystem configure max threads and core threads independently
+:author:            Cheng Fang
+:email:             cfang@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-10057 (EJB subsystem configure max threads and core threads independently)
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-488 (Fine-grained control on ejb3 thread-pool and reuse idle threads in pool before creating new)
+* https://issues.jboss.org/browse/WFCORE-1446 (EJB thread pool keepalive not honored and is not reusing inactive threads)
+
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:szhantem@redhat.com[Sultan Zhantemirov]
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+[x] Engineering
+
+[x] QE
+
+=== Affected Projects or Components
+
+* ejb3
+* wildfly-core/threads
+
+=== Other Interested Projects
+
+== Requirements
+
+=== Hard Requirements
+
+The following table shows the new requirements vs the existing behavior in ejb3 subsystem thread-pool:
+
+
+|===
+|New Requirement |Existing Behavior of EJB3 Subsystem thread-pool
+
+|Ability to configure core-threads and max-threads independently
+|max-threads is configurable, but core-threads is not configurable and always equals to max-threads
+
+|Ability to reuse available threads as much as possible, without unnecessary creation of new threads
+|Upon new request, new threads are created up to the limit of max-threads, even though idle threads are available
+
+|Ability to timeout idle non-core threads after keepalive-timeout
+|Idle threads are not timed out, and keepalive-timeout is ignored
```

Comment:
@chengfang There is `100 ms` by default set for `keepalive-timeout` in current implementation. Do we have any recommendation to which value it should be set? I think that current default is very low as for example for Artemis thread pools there was keepalive timeout set to `60 seconds`.